pkg/build/buildutil/util.go

```diff
@@ -275,3 +276,32 @@ func LabelValue(name string) string {
 	klog.Warningf("In creating the value of the build label in the build pod, several attempts at manipulating %s to meet k8s label name requirements failed", name)
 	return name
 }
+
+func fromMatch(from1 k8stypes.NamespacedName, from2 k8stypes.NamespacedName) bool {
+	return from1.Namespace == from2.Namespace && from1.Name == from2.Name
+}
+
+// GetImageChangeTriggerStatusForImageChangeTrigger returns the ImageChangeTrigger entry in status that corresponds
+// to the supplied ImageChangeTrigger entry in the spec.
+func GetImageChangeTriggerStatusForImageChangeTrigger(triggerSpec *buildv1.ImageChangeTrigger, bc *buildv1.BuildConfig) *buildv1.ImageChangeTriggerStatus {
+	if bc.Status.ImageChangeTriggers == nil || len(bc.Status.ImageChangeTriggers) == 0 {
+		return nil
+	}
+	specFrom := k8stypes.NamespacedName{}
```

Comment:
@cdaley - since last time, building `specFrom` did not need to be in the for loop on `bc.Status.ImageChangeTriggers`